services/subscriber/service.go

```diff
@@ -38,53 +40,50 @@ type subEntry struct {
 // to defined third party destinations.
 // Subscriptions are defined per database and retention policy.
 type Service struct {
-	subs       map[subEntry]PointsWriter
 	MetaClient interface {
 		Databases() []meta.DatabaseInfo
 		WaitForDataChanged() chan struct{}
 	}
 	NewPointsWriter func(u url.URL) (PointsWriter, error)
 	Logger          *log.Logger
+	update          chan struct{}
 	statMap         *expvar.Map
 	points          chan *coordinator.WritePointsRequest
 	wg              sync.WaitGroup
 	closed          bool
 	closing         chan struct{}
 	mu              sync.Mutex
+	conf            Config
 }
 
 // NewService returns a subscriber service with given settings
 func NewService(c Config) *Service {
-	return &Service{
-		subs:            make(map[subEntry]PointsWriter),
-		NewPointsWriter: newPointsWriter,
-		Logger:          log.New(os.Stderr, "[subscriber] ", log.LstdFlags),
-		statMap:         influxdb.NewStatistics("subscriber", "subscriber", nil),
-		points:          make(chan *coordinator.WritePointsRequest),
-		closed:          true,
-		closing:         make(chan struct{}),
+	s := &Service{
+		Logger:  log.New(os.Stderr, "[subscriber] ", log.LstdFlags),
+		statMap: influxdb.NewStatistics("subscriber", "subscriber", nil),
+		points:  make(chan *coordinator.WritePointsRequest, 100),
+		closed:  true,
+		conf:    c,
 	}
+	s.NewPointsWriter = s.newPointsWriter
+	return s
 }
 
 // Open starts the subscription service.
 func (s *Service) Open() error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-
-	s.closing = make(chan struct{})
-
 	if s.MetaClient == nil {
-		panic("no meta store")
+		return errors.New("no meta store")
 	}
 
 	s.closed = false
 
-	// Perform initial update
-	s.Update()
+	s.closing = make(chan struct{})
+	s.update = make(chan struct{})
 
-	s.wg.Add(1)
-	go s.writePoints()
-	// Do not wait for this goroutine since it block until a meta change occurs.
+	s.wg.Add(2)
+	go s.run()
```

Comment:
It's just a personal preference, but I like to see the `Done()` in close proximity to the `Add()` for `WaitGroup`s. Rather than seeing them separated by pages of code in different functions.

What do you think about `go func() { defer wg.Done(); s.run() }()` instead of embedding the `wg.Done()` in each of the called functions?